ipaserver/plugins/idrange.py

```diff
@@ -243,8 +242,8 @@ class idrange(LDAPObject):
         StrEnum('iparangetype?',
                 label=_('Range type'),
                 cli_name='type',
-                doc=(_('ID range type, one of {vals}'
-                     .format(vals=', '.join(sorted(range_types))))),
+                doc=_('ID range type, one of ' +
+                      'ipa-local, ipa-ad-trust, ipa-ad-trust-posix'),
```

Comment:
Why drop the generated list in favor of a hardcoded one?

Reply:
1) format() triggers real translation. Such methods should be used for translatable objects at request time only. Otherwise message will be at server locale.

Concerning to Web UI.
On execution of json_metadata command idrange object is transformed into json by json_serialize. Which calls ```__json__``` method to Gettext / NGettext instances and the translation is performed. 
In case of _('{vals}').format - idrange stores already translated message and the carriage turns into a pumpkin.

2) unfortunately, it's impossible to specify 
```python
doc=_('ID range type, one of ' + ', '.join(sorted(range_types)))
```
output po template is 
```python
'ID range type, one of , '
```

IMO. the only solution for now.

Reply:
Please replace it with
```
doc=(_('ID range type, one of {vals}').
         .format(vals=', '.join(sorted(range_types)))),
```
This way you'd have the string for translation with a formatting context and we'll fill up the values at runtime. You don't need to translate range types as they are option values.

Reply:
I agree that range_types should be not translatable.
But as i said before it is not runtime context.
Please, take a look:

Even this message is here
```shell
msgunfmt /usr/share/locale/ru/LC_MESSAGES/ipa.mo | grep 'ID range type, one of {vals}' -A 1
msgid "ID range type, one of {vals}"
msgstr "Тип диапазона идентификаторов, одно из следующих значений: {vals}"
```
There is no translation in Web UI:
![image](https://user-images.githubusercontent.com/31205609/41642134-2252bb24-7470-11e8-97c2-4ee894971c37.png)

Reply:
I see what you are getting to. I think we should just drop the second part here and use
```
doc=_("ID range type, one of allowed values")
```
In both Web UI API browser and in `ipa idrange-add --help` we get the list of allowed values anyway so there is no loss of context.